stackrox/acs-fleet-manager
# ROX-23219: Specify additional tenant's auth via gitops

## Description

This PR introduces ability to configure additional auth provider for ACSCS instance in addition to default `RHSSO` auth provider. The configuration is done via gitops config.
This feature is introduced because our default auth provider can be too restrictive for multi-organization customers. For these customers, SREs/oncall engineers can configure additional auth provider that would not have such restrictions.

Implementation details:
1. The only supported type of auth provider at the moment is OIDC
2. If OIDC clientID is not specified, client credentials from the default `RHSSO` auth provider are reused.

## Checklist (Definition of Done)


## Test manual

1. Deploy locally
2. Create new central instance
3. Add next section to `gitops-config.yaml` `centrals` value:
```
  additionalAuthProviders:
    - instanceId: "<instance id>"
      authProvider:
        name: "good-name"
        groups:
          - key: "a"
            value: "b"
            role: "Admin"
```
4. Check that `cloud-service-sensible-declarative-configs` secret got 3rd value inside - additional auth provider declarative config
